utils/misc/misc.go

```diff
@@ -276,13 +277,30 @@ func UnZipSingleFile(outputfile string, filename string) {
 	rc.Close()
 }
 
-func RemoveFilePaths(filepaths ...string) {
-	for _, filepath := range filepaths {
+// RemoveFilePaths removes filePaths as well as cleans up the empty folder structure.
+func RemoveFilePaths(filePaths ...string) {
+	for _, filepath := range filePaths {
 		err := os.Remove(filepath)
 		if err != nil {
 			pkgLogger.Error(err)
 		}
 	}
+	removeEmptyFolderStructure(filePaths...)
+}
+
+// removeEmptyFolderStructure recursively cleans up everything till it reaches the stage where the folders are not empty or parent.
+func removeEmptyFolderStructure(filepaths ...string) {
```

Comment:
Why do we need filepaths to be `...string`. It looks like we are only using `filepaths[0]` anyways.

Reply:
Removed ...